src/test/java/bio/terra/workspace/service/workspace/flight/CreateGoogleContextFlightTest.java

```diff
@@ -129,6 +136,21 @@ private static FlightMap createInputParameters(
     return inputs;
   }
 
+  /**
+   * Asserts that a provided project has every custom role specified in {@link
+   * CustomGcpIamRoleMapping}
+   */
+  private void assertRolesExist(Project project) throws IOException {
+    for (CustomGcpIamRole customRole : CustomGcpIamRoleMapping.customIamRoles) {
+      String fullRoleName =
+          "projects/" + project.getProjectId() + "/roles/" + customRole.getRoleName();
+      Role gcpRole = crl.getIamCow().projects().roles().get(fullRoleName).execute();
+      assertThat(
+          gcpRole.getIncludedPermissions(),
+          containsInAnyOrder(customRole.getIncludedPermissions().toArray()));
+    }
+  }
+
```

Comment:
Can you add some tests to verify the transience of Writer containing reader roles, etc?

Reply:
I don't think this is a requirement, so I don't want to enforce this in tests. Right now we're pretty sure we want that structure, but that could easily change and won't be true for other roles, e.g. editor,

Reply:
ok